ReactiveCocoa/Swift/Flatten.swift

```diff
@@ -459,20 +459,56 @@ private final class ConcatState<Value, Error: Swift.Error> {
 
 	/// Subscribes to the given signal producer.
 	func startNextSignalProducer(_ signalProducer: SignalProducer<Value, Error>) {
-		signalProducer.startWithSignal { signal, disposable in
-			let handle = self.disposable?.add(disposable) ?? nil
-
-			signal.observe { event in
-				switch event {
-				case .completed, .interrupted:
-					handle?.remove()
-
-					if let nextSignalProducer = self.dequeueSignalProducer() {
-						self.startNextSignalProducer(nextSignalProducer)
+		// Are we still inside the loop
+		var isLooping = true
+		// Current signal producer
```

Comment:
These comments aren't really helpful and should be removed. They're just restating what the code already tells us, rather than providing explanations/background.